src/fondant/data_io.py

```diff
@@ -159,7 +163,7 @@ def __init__(
     def write_dataframe(self, dataframe: dd.DataFrame) -> None:
         write_tasks = []
 
-        dataframe.index = dataframe.index.rename("id").astype("string")
+        dataframe.index = dataframe.index.rename("id")
```

Comment:
An integer columns is no longer sorted if we convert it to string. We should keep the index sorted.

Reply:
No but I don't think there's a reason for us to cast it? We can just keep the type.

Reply:
Makes sense, I thought for some reason that we had a fixed schema for the index at the ComponentSpec level but that's not the case